R/hospitalizations_last_n_days.R

```diff
@@ -0,0 +1,112 @@
+#' @title Compute number of previous hospitalizations for an encounter in a
+#'   given window.
+#'
+#' @description Computes the number of hospitalizations an encounter has had
+#'   over a specified time period, using admission & discharge date-times from
+#'   the CIHI Discharge Abstract Database (DAD).
+#'
+#' @details This function uses DAD fields Admission date-time (Group 04 Fields
+#'   01/02), Discharge date-time (Group 05 Fields 01/02) along with
+#'   patient_id_hashed to calculate the number of previous hospitalizations an
+#'   encounter has had in a specified window.
+#'
+#' @param cohort (`data.table`, `data.frame`)\cr Cohort table with all relevant
+#'   encounters of interest, where each row corresponds to a single encounter.
+#'   Must contain the following columns:
+#' - `genc_id` (`integer`): GEMINI encounter ID
+#' - `patient_id_hashed` (`Varchar`): Unique hash of patient's health card number
+#' - `admission_date_time` (`character`): The date and time that the patient was officially registered as an inpatient for the current hospital admission. Must be in YYYY-MM-DD HH:MM.
+#' - `discharge_date_time` (`character`): The date and time when the patient was formally discharged from the current hospital admission. Must be in YYYY-MM-DD HH:MM.
+#'
+#'
+#' @param n_days (`integer`)\cr Window for calculating the number of
+#'   hospitalizations. n_days is set to 30 days by default.
+#'
+#' @param admit_dt (`character`)\cr Admission date-time Variable representing
+#'   admission date-time. Set to admission_date_time by default.
+#'
+#' @param discharge_dt Discharge date-time Variable representing discharge
+#'   date-time. Set to discharge_date_time by default.
+#'
+#' @return (`data.table`)\cr data.table containing the same number of rows as
+#'   input `cohort` table. This data.table contains the genc_id, along with
+#'   `n_hospitalizations`, which represents the number of previous
+#'   hospitalizations in the given window. The values in `n_hospitalizations`
+#'   will be 0, 1, or 2+.
+#'
+#' @export
+#'
+#' @examples
+#' # Default n_days of 30:
+#' \dontrun{
+#' derive_n_hosp_prev(cohort)
```

Comment:
`derive_n_hosp_prev(cohort)` should be changed to `hospitalizations_last_n_days(cohort)`